### PR TITLE
Update docker documentation

### PR DESCRIPTION
**Pull request by DHager**
Markdown-only changes. 

* Replace a broken `docker-compose exec` command with a general access to the terminal
* Removes a `docker-compose build` step which currently does nothing
* Adds instructions for doing commits (and more-importantly commit-*hooks*) inside the container
* Documents a failure scenario where a Docker-VM doesn't have enough memory

### CODE REVIEW

File: docs/Docker.md [hunk 1]
```diff
@@ -21,9 +21,48 @@ Run the following commands from the DIM cloned directory on your machine:
 * `ctrl+c` to stop
 * `docker-compose up -d` to start in detached mode
 * `docker-compose stop` to stop detached mode
-* `docker-compose build` to re-build if the compose files change
 
-### Tips and How-To's
+### Tips and Troublehsooting
```
Comment (delphiactual) [inline docs/Docker.md:25]:
```suggestion
### Tips and Troubleshooting
```

File: docs/Docker.md [hunk 2]
```diff
@@ -21,9 +21,48 @@ Run the following commands from the DIM cloned directory on your machine:
...
+## Commit hook won't run
+
+If you `git commit` inside the container, you may see:
+
+    Can't find Husky, skipping pre-commit hook
+    You can reinstall it using 'npm install husky --save-dev' or delete this hook
+    Can't find Husky, skipping prepare-commit-msg hook
+    You can reinstall it using 'npm install husky --save-dev' or delete this hook
+
+This can be fixed with the suggested commands. A
```
Comment (delphiactual) [inline docs/Docker.md:38]: Looks like you started another sentence.

Comment (DHager) [reply docs/Docker.md:38]: Whoops...

File: docs/Docker.md [hunk 3]
```diff
@@ -21,9 +21,48 @@ Run the following commands from the DIM cloned directory on your machine:
...
+
+## Commit hook won't run
+
+If you `git commit` inside the container, you may see:
+
+    Can't find Husky, skipping pre-commit hook
+    You can reinstall it using 'npm install husky --save-dev' or delete this hook
+    Can't find Husky, skipping prepare-commit-msg hook
+    You can reinstall it using 'npm install husky --save-dev' or delete this hook
+
```
Comment (delphiactual) [inline docs/Docker.md:37]: I don't think we should recommend removing hooks, as they are where linting occur etc.

File: docs/Docker.md [hunk 4]
```diff
@@ -21,9 +21,48 @@ Run the following commands from the DIM cloned directory on your machine:
...
+For an interactive editor, try:
+
+    apt-get update
+    apt-get install nano
+
+This will install the `nano` editor. Another option is `vim`.
+
+## Node modules
+
+If you need to install new node dependencies, you should run those commands from inside the container. Once inside, use `npm install NameOfTheDependency` as normal.
```
Comment (delphiactual) [inline docs/Docker.md:60]:
```suggestion
If you need to install new node dependencies, you should run those commands from inside the container. Once inside, use `yarn install NameOfTheDependency` as normal.
```

File: docs/Docker.md (hunk 3 above)
Comment (DHager) [reply docs/Docker.md:37]: I think that's outside our direct/easy control, it's [from the husky tool](https://github.com/typicode/husky/blob/4e1d440be7e3058036d24e7c5558c09448772af2/sh/husky.sh#L20).

**Pull request merged**